### PR TITLE
Update to cdaweb list_remote_files

### PR DESCRIPTION
**Pull request by jklenzing**
Addresses #200.

This has been fixed for a while in the maven_ngims branch.  Full implementation there is dependent on other file issues (non-unique datetimes in timeseries) in ngims, which may take a while to sort out.  Requesting a pull on this now to make the cdaweb upgrade cleaner and close out issue #200.

### CODE REVIEW

**Comment by rstoneback** [comment]
Thanks Jeff. I’ll resume pysat activities after an upcoming telecon and lunch.

Cheers,
Russell


On Jun 13, 2019, at 6:40 AM, Jeff Klenzing <notifications@github.com<mailto:notifications@github.com>> wrote:


@jklenzing<https://github.com/jklenzing> requested your review on: #238<https://github.com/rstoneback/pysat/pull/238> Update to cdaweb list_remote_files.

File: pysat/instruments/methods/nasa_cdaweb.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def list_files(tag=None, sat_id=None, data_path=None, format_str=None,
             try:
                 format_str = supported_tags[sat_id][tag]
             except KeyError:
-                raise ValueError('Unknown tag')
+                raise ValueError('Unknown tag: ' + tag)
```
Comment (rstoneback) [inline pysat/instruments/methods/nasa_cdaweb.py]: I suppose a bad sat_id could also cause a KeyError as well. How about a unknown sat_id and tag combo error statement?

File: pysat/instruments/methods/nasa_cdaweb.py [hunk 2]
```diff
@@ -285,6 +291,7 @@ def download(supported_tags, date_array, tag, sat_id,
                                 date.strftime('%d %B %Y'))))
                 sys.stdout.flush()
                 remote_files = list_remote_files(tag=tag, sat_id=sat_id,
+                                                 remote_site=remote_site,
```
Comment (rstoneback) [inline pysat/instruments/methods/nasa_cdaweb.py]: Do we have a potential issue now that times within a day are included in the formatted names but not in the list_remote_files method?

Comment (rstoneback) [reply pysat/instruments/methods/nasa_cdaweb.py]: github was hiding code and I initially missed we are in a single file per day situation.

File: pysat/instruments/methods/nasa_cdaweb.py (hunk 1 above)
Comment (jklenzing) [reply pysat/instruments/methods/nasa_cdaweb.py]: Will update

Comment (jklenzing) [reply pysat/instruments/methods/nasa_cdaweb.py]: fixed

File: pysat/instruments/methods/nasa_cdaweb.py (hunk 2 above)
Comment (jklenzing) [reply pysat/instruments/methods/nasa_cdaweb.py]: That's right, this bit is only for multi-file days

**Pull request merged**